run_bot_nix.sh

```
@@ -69,20 +83,42 @@ create_output_dir() {
 # Process JSON configuration to add UUID if missing
 process_config() {
     local config_json=$1
+    local use_api_mode=$2
     local bot_uuid=$(generate_uuid)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)